### PR TITLE
lib/storage/partition: prevent panic in case resulting in-memory part is empty after merge

### PR DESCRIPTION
**Pull request by zekker6**
### Describe Your Changes

It is possible for in-memory part to be empty if ingested samples are removed by retention filters. In this case, data will not be discarded due to retention before creating in memory part. After in-memory parts merge samples will be removed resulting in creating completely empty part at destination.

### Checklist

The following checks are **mandatory**:

- [x] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

File: lib/storage/partition.go [hunk 1]
```diff
@@ -806,6 +809,11 @@ func (pt *partition) mustMergeInmemoryPartsFinal(pws []*partWrapper) *partWrappe
 	}
 	mpDst.ph = *ph
 
+	// resulting part is empty, no need to create a part wrapper
+	if ph.BlocksCount == 0 {
```
Comment (f41gh7) [inline lib/storage/partition.go:813]: Could we add test for this case somehow?

Comment (zekker6) [reply lib/storage/partition.go:813]: Sure, added a test to cover this case

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks for contribution!

**Pull request merged**